### PR TITLE
Kh dev admin dashboard

### PR DESCRIPTION
**Pull request by kheppenstall**
-Add feature that admin can view all orders
-Add feature that admin can filter order by status
-Extract orders part of admin dashboard into partial

NOTE: Does not close card 113 or story 18. Still need to add functionality for an admin to edit the status of an order.

### CODE REVIEW

**Comment by kheppenstall** [comment]
@akintner @tmikeschu @wlffann Ready for review. Needs all three.

File: app/views/admin/_admin-orders.html.haml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.orders-container
+  .orders-header
+    .header
+      %h3 #{@header if @header} Orders
+    .sort-by
+      .dropdown
+        %a.dropdown-toggle{ "data-toggle" => "dropdown", href: "#" }
```
Comment (tmikeschu) [inline app/views/admin/_admin-orders.html.haml]: Killer logic!

File: spec/features/admin/admin_views_orders_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+RSpec.feature do
+  before do
+    @ordered_orders = create_list(:order_with_items, 2, status: 0)
+    @paid_orders = create_list(:order_with_items, 2, status: 1)
+    @cancelled_orders = create_list(:order_with_items, 2, status: 2)
+    @completed_orders = create_list(:order_with_items, 2, status: 3)
+    @orders = Order.all
+    user = create(:user, role: 1)
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
```
Comment (akintner) [inline spec/features/admin/admin_views_orders_spec.rb]: Like the before block for this test.

File: spec/features/admin/admin_views_orders_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        end
+      end
+
+      scenario 'filter orders by status' do
+        Order.statuses.keys.each do |status|
+          click_on status
+          expect(current_path).to eq admin_dashboard_path
+          expect(page).to have_content "#{status.capitalize} Orders"
+          expect_only_orders("#{status}?")
+        end
```
Comment (wlffann) [inline spec/features/admin/admin_views_orders_spec.rb]: That's a good looking test!

**Comment by akintner** [comment]
@kheppenstall Once the merge conflict is fixed, I think this is good to go.

File: spec/features/admin/admin_views_orders_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          expect(page).to have_link 'Order Details', href: order_path(order)
+        end
+      end
+
+      scenario 'filter orders by status' do
+        Order.statuses.keys.each do |status|
+          click_on status
+          expect(current_path).to eq admin_dashboard_path
+          expect(page).to have_content "#{status.capitalize} Orders"
+          expect_only_orders("#{status}?")
```
Comment (tmikeschu) [inline spec/features/admin/admin_views_orders_spec.rb]: Nice helper method. So clean!

**Comment by wlffann** [comment]
@kheppenstall It's looking good to me

File: app/models/order.rb [hunk 5]
```diff
@@ -31,4 +31,11 @@ def checkout(cart)
     end
   end
 
+  def self.by_status(status)
+    if Order.statuses.keys.include?(status)
```
Comment (tmikeschu) [inline app/models/order.rb]: Can we clean this up some how with more AR-y methods? The hash chain seems a little hacky to me. @kheppenstall

**Pull request merged**